### PR TITLE
test/lazy-omap-stats: regex optimize for header index find

### PR DESCRIPTION
**Pull request by NitzanMordhai**
With the current regex, and large output from pg dump, regex_search can sometimes causing a segfault.
by changing the header regex to find less intensive regex the regex_search won't segfault.

Fixes: https://tracker.ceph.com/issues/59196

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by NitzanMordhai** [comment]
jenkins test make check

**Comment by NitzanMordhai** [comment]
@badone can you please review?

File: src/test/lazy-omap-stats/lazy_omap_stats_test.cc [hunk 1]
```diff
@@ -563,10 +563,7 @@ void LazyOmapStatsTest::wait_for_active_clean()
   cout << "Waiting for active+clean" << endl;
 
   int index = -1;
-  regex reg(
-      "\n"
-      R"((PG_STAT[\s\S]*))"
-      "\n +\n[0-9]");
+  regex reg(R"(PG_STAT[^\n]*\n)");
   string command = R"({"prefix": "pg dump"})";
```
Comment (badone) [inline src/test/lazy-omap-stats/lazy_omap_stats_test.cc:567]: Of course these are not equivalent and the change should not be necessary but, if you are confident this covers all bases then I'm all for it.

**Review by badone — APPROVED** [review]

**Comment by sseshasa** [comment]
**Teuthology test result**

@NitzanMordhai There was a related failure in https://pulpito.ceph.com/yuriw-2024-03-08_16:20:46-rados-wip-yuri4-testing-2024-03-05-0854-distro-default-smithi/.

**Related Failures**

The fix worked partially but failed down the line at some other point in the script.
1. https://pulpito.ceph.com/yuriw-2024-03-08_16:20:46-rados-wip-yuri4-testing-2024-03-05-0854-distro-default-smithi/7587943
2. https://pulpito.ceph.com/yuriw-2024-03-08_16:20:46-rados-wip-yuri4-testing-2024-03-05-0854-distro-default-smithi/7587684

For the full report please see: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcQFtoIRXE1971-wip-yuri4-testing-2024-03-05-0854

**Comment by badone** [comment]
The failing tests were using the older version of the patch with the change only to  LazyOmapStatsTest::wait_for_active_clean() (9e0ca2755b1fc7f3b78de9816f3ce1a60f528f59) and no change to LazyOmapStatsTest::check_one() which is where the test is failing. Can we try again with the patch as it currently stands?

**Comment by amathuria** [comment]
Saw a segfault in the test run including this PR - https://pulpito.ceph.com/yuriw-2024-03-19_00:09:45-rados-wip-yuri5-testing-2024-03-18-1144-distro-default-smithi/7609959/ in the ceph_test_lazy_omap_stats test. 

PRs being tested in this batch are noted here: https://tracker.ceph.com/issues/64974

**Comment by badone** [comment]
> Saw a segfault in the test run including this PR - https://pulpito.ceph.com/yuriw-2024-03-19_00:09:45-rados-wip-yuri5-testing-2024-03-18-1144-distro-default-smithi/7609959/ in the ceph_test_lazy_omap_stats test.
> 
> PRs being tested in this batch are noted here: https://tracker.ceph.com/issues/64974

This approach is just going to be whack-a-mole I'm afraid due to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86164 I'm looking into an alternative approach and I hope that's OK with you @NitzanMordhai ?

**Comment by NitzanMordhai** [comment]
> This approach is just going to be whack-a-mole I'm afraid due to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86164 I'm looking into an alternative approach and I hope that's OK with you @NitzanMordhai ?

@badone sure, thanks a lot for picking it up!

**Comment by badone** [comment]
Closing in favour of https://github.com/ceph/ceph/pull/55596 Thank you for your effort Nitzan.

**Pull request closed without merging**